CI failure looks real:

```
___________________________________ test_pow ___________________________________
[gw0] linux -- Python 3.10.13 /opt/hostedtoolcache/Python/3.10.13/x64/bin/python
scipy/spatial/transform/tests/test_rotation.py:1596: in test_pow
    assert np.all(ang < atol)
E   assert False
E    +  where False = <function all at 0x7f8ce436b530>(array([2.97024011, 0.26148642, 1.19333715, 1.43570018, 2.02521876,\n       1.04586757, 0.27726364, 0.70317966, 0.28400547, 0.67914956]) < 1e-14)
E    +    where <function all at 0x7f8ce436b530> = np.all
        _          = 1
        ang        = array([2.97024011, 0.26148642, 1.19333715, 1.43570018, 2.02521876,
       1.04586757, 0.27726364, 0.70317966, 0.28400547, 0.67914956])
        atol       = 1e-14
        n          = -2
        p          = <scipy.spatial.transform._rotation.Rotation object at 0x7f8cb9de3e10>
        p_inv      = <scipy.spatial.transform._rotation.Rotation object at 0x7f8cb9de3960>
        q          = <scipy.spatial.transform._rotation.Rotation object at 0x7f8cb9de3780>
        r          = <scipy.spatial.transform._rotation.Rotation object at 0x7f8cb9de34b0>
```

--- a/scipy/spatial/transform/_rotation.pyx
+++ b/scipy/spatial/transform/_rotation.pyx
@@ -2642,11 +2642,14 @@ cdef class Rotation:
 
         # Exact short-cuts
         if n == 0:
-            return Rotation.identity(len(self._quat))
+            return Rotation.identity(None if self._single else len(self._quat))
         elif n == -1:
             return self.inv()
         elif n == 1:
-            return self.__class__(self._quat.copy())
+            if self._single:
+                return self.__class__(self._quat[0], copy=True)
+            else:
+                return self.__class__(self._quat, copy=True)
         else:  # general scaling of rotation angle
             return Rotation.from_rotvec(n * self.as_rotvec())
 

--- a/scipy/spatial/transform/tests/test_rotation.py
+++ b/scipy/spatial/transform/tests/test_rotation.py
@@ -1584,6 +1584,7 @@ def test_pow():
     p_inv = p.inv()
     # Test the short-cuts and other integers
     for n in [-5, -2, -1, 0, 1, 2, 5]:
+        # Test accuracy
         q = p ** n
         r = Rotation.identity(10)
         for _ in range(abs(n)):
@@ -1593,6 +1594,12 @@ def test_pow():
                 r = r * p_inv
         ang = (q * r.inv()).magnitude()
         assert np.all(ang < atol)
+
+        # Test shape preservation
+        r = Rotation.from_quat([0, 0, 0, 1])
+        assert (r**n).as_quat().shape == (4,)
+        r = Rotation.from_quat([[0, 0, 0, 1]])
+        assert (r**n).as_quat().shape == (1, 4)
 
     # Large angle fractional
     for n in [-1.5, -0.5, -0.0, 0.0, 0.5, 1.5]:

